api/lib/domain/usecases/remove-authentication-method.js

```diff
@@ -2,9 +2,13 @@ import { NON_OIDC_IDENTITY_PROVIDERS } from '../constants/identity-providers.js'
 import * as OidcIdentityProviders from '../constants/oidc-identity-providers.js';
 import { UserNotAuthorizedToRemoveAuthenticationMethod } from '../errors.js';
 
-const removeAuthenticationMethod = async function ({ userId, type, userRepository, authenticationMethodRepository }) {
+export const removeAuthenticationMethod = async function ({
+  userId,
+  type,
```

Comment:
Ne devrait-on pas renommer cette variable plutôt en `identityProvider` ou `identityProviderCode` ?

Reply:
~Tout à fait ! Bien vu ! Je suis pour l'utilisation de `identityProviderCode` qui est le plus explicite :~

```suggestion
  identityProviderCode,
```

Reply:
Ah non, en fait `identityProviderCode` n'est pas adapté car il y a `EMAIL` et `USERNAME` qui ne sont pas des identity providers !

Reply:
Alors `authenticationMethodType` pourrait être plus précis.

Reply:
D'accord pour `authenticationMethodType` !

Reply:
C'est fait !